File: 백준/Silver/17615. 볼 모으기/볼 모으기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
	
	static char[] line;

	public static void main(String[] args) throws NumberFormatException, IOException {

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int N = Integer.parseInt(br.readLine());
		line = br.readLine().toCharArray();
		int min = Integer.MAX_VALUE;
		
		min = Math.min(checkLeft('R', 0, N), min);
		min = Math.min(checkLeft('B', 0, N), min);
		min = Math.min(checkRight('R', 0, N), min);
		min = Math.min(checkRight('B', 0, N), min);
		
		System.out.println(min);
	}
	
	static int checkLeft(char target, int start, int end) {
		int res = 0, cnt = 0;
		for (int i = start; i < end; i ++) {
			if (line[i] != target) {
				cnt = 1;
			} else {
				res += cnt;
			}
		}
		return res;
	}
	
	static int checkRight(char target, int start, int end) {
		int res = 0, cnt = 0;
		for (int i = end - 1; i >= start; i--) {
			if (line[i] != target) {
				cnt = 1;
			} else {
				res += cnt;
			}
		}
		return res;
	}

}

File: 백준/Silver/17615. 볼 모으기/볼 모으기.py
import sys
input = sys.stdin.readline


def check(target, start, end, dir):
    res = cnt = 0
    i = start if dir == 'LEFT' else end - 1
    limit = end if dir == 'LEFT' else start
    step = 1 if dir == 'LEFT' else -1

    while (i < limit) if dir == 'LEFT' else (i >= limit):
        if line[i] != target:
            cnt = 1
        else:
            res += cnt
        i += step

    return res


N = int(input())
line = list(input().rstrip())

ans = 500001
ans = min(check('R', 0, N, 'LEFT'),
          check('B', 0, N, 'LEFT'),
          check('R', 0, N, 'RIGHT'),
          check('B', 0, N, 'RIGHT'),
          ans)
print(ans)